src/bin/common/solving_loop.ml

```diff
@@ -534,33 +535,15 @@ let main () =
   in
 
   let set_sat_solver sat st =
-    let optim = DO.Optimize.get st in
-    match sat with
-    | Util.Tableaux | Tableaux_CDCL when optim ->
-      warning
-        "Sat-solver %a is incompatible with optimization: ignoring command."
-        Util.pp_sat_solver
-        sat;
-      st
-    | Tableaux | Tableaux_CDCL | CDCL | CDCL_Tableaux ->
-      O.set_sat_solver sat;
-      (* `make_sat` returns the sat solver corresponding to the new sat_solver
-         option. *)
-      DO.SatSolver.set sat st
+    O.set_sat_solver sat;
+    (* `make_sat` returns the sat solver corresponding to the new sat_solver
+       option. *)
+    DO.SatSolver.set sat st
   in
 
-  let set_optimize optim st =
-    let sat = DO.SatSolver.get st in
-    match sat with
-    | Util.Tableaux | Tableaux_CDCL when optim ->
-      warning
-        "Sat-solver %a is incompatible with optimization: ignoring command."
-        Util.pp_sat_solver
-        sat;
-      st
```

Comment:
Just a remainder. As we don't explore different SAT branches in `FunSAT` and we don't force the SAT solver to be `SatML` in presence of the MaxSMT extension, it means the feature is broken in presence of `--sat-solver tableaux` for instance. We should mention it at least in the documentation of AE for the next release.

Reply:
The Tableaux solver errors out if it receives an optimization command (see implementation of `Fun_sat_frontend.optimize`).

Reply:
Uhm I'm not sure we should crash with an error here. We don't produce wrong model but we don't produce the best one (according to our definition of best one).

Reply:
I agree that this is a bad error mechanism, but I'd argue that returning an error here is appropriate — the user is requesting an optimized model, but we can't satisfy that request. I'll change the mechanism when we encounter an `optimize` in the Tableaux solver.